lib/ansible/parsing/mod_args.py

```diff
@@ -27,6 +27,9 @@
 from ansible.parsing.splitter import parse_kv, split_args
 from ansible.template import Templar
 
+# dynamically installed roles ...
+from ansible.galaxy.helpers import get_role_by_module_fqn
```

Comment:
Seems wrong to have mod_args import galaxy code just to look a module file name for a galaxy role nickname.

Maybe remove the galaxy bits from here, and instead populate the 'jctanner.pinger.pinger' style name as a name or alias in plugins.PluginLoader()? ie, add 'jctanner.pinger.pinger' to the names in the cache in https://github.com/ansible/ansible/blob/unified_ansible/lib/ansible/plugins/__init__.py#L273-L285

Then this module doesn't need to know it's a galaxy specific name.  

(the name could be URI like with a scheme indicating it's a galaxy name 'galaxy_role_module://jctanner/pinger/module/pinger' or whatever. Someway for mod args to tell that is referring to a module in a galaxy role.)